src/components/Dialog.tsx

```diff
@@ -14,57 +20,47 @@ type Props = {
   keyUp?: React.KeyboardEventHandler<HTMLDivElement>;
 };
 
-export default class Dialog extends Component<Props> {
-  titleRef: React.Ref<unknown>;
-  constructor(props: Props) {
-    super(props);
-    this.titleRef = React.createRef();
-  }
+const Dialog = (props: Props) => {
+  const { appElement, isOpen, closeFn, keyUp, body } = props;
+  var headerRef = useRef<HTMLHeadingElement>(null);
 
-  static propTypes = {
-    appElement: PropTypes.object.isRequired,
-    isOpen: PropTypes.bool.isRequired,
-    closeFn: PropTypes.func.isRequired,
-    keyUp: PropTypes.func,
-    body: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]).isRequired,
-  };
+  useEffect(() => {
+    Modal.setAppElement(appElement);
+  }, []);
 
-  onKeyUp = (e: KeyboardEvent) => {
-    if (e.key === "Escape") this.props.closeFn();
+  const onKeyUp = (e: KeyboardEvent): void => {
+    if (e.key === "Escape") closeFn();
   };
 
-  componentDidMount() {
-    Modal.setAppElement(this.props.appElement);
-  }
+  // NOTE(Emmanuel): this feels like a level of indirection
+  // that refs are supposed to deal with.
+  const focusTitle = () => {
+    headerRef.current.focus();
+  };
 
-  focusTitle() {
-    document.getElementById("heading").focus();
-  }
+  return (
+    <Modal
+      isOpen={isOpen}
+      className={"madeUpClassToCancelDefaultStyles"}
+      onRequestClose={closeFn}
+      shouldCloseOnEsc={true}
+      shouldReturnFocusAfterClose={true}
+      shouldFocusAfterRender={true}
+      shouldCloseOnOverlayClick={true}
+      onAfterOpen={focusTitle}
+      aria={{ labelledby: "heading" }}
+    >
+      <div tabIndex={-1} id="DialogContents" onKeyUp={keyUp || onKeyUp}>
```

Comment:
is there any reason to keep `id="DialogContents"` around?

Reply:
also prefer `onKeyUp={keyUp ?? onKeyUp}`

Reply:
`#DialogContents` is used by our CSS rules. Switched `||` to `??`